### PR TITLE
Integrate 3/18 nightly RN build.

### PR DESCRIPTION
**Pull request by rectified95**
Commits: https://github.com/facebook/react-native/compare/d477f8011...cfb5b5e20

Closes https://github.com/microsoft/react-native-windows/issues/7309 by pulling in the updated `MapBuffer.cpp`
Closes https://github.com/microsoft/react-native-windows/issues/7289 by pulling in an update to `Yoga.cpp`

Issues discovered during this integration, not necessarily caused by this batch, though still recent regressions:
https://github.com/microsoft/react-native-windows/issues/7557
https://github.com/microsoft/react-native-windows/issues/7556
https://github.com/microsoft/react-native-windows/issues/7555
https://github.com/microsoft/react-native-windows/issues/7554

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7549)

### CODE REVIEW

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js [hunk 1]
```diff
@@ -26,9 +26,24 @@ const ComponentExamples: Array<RNTesterExample> = [
...
+  {
+    key: 'FlatList-onEndReached',
+    module: require('../examples/FlatList/FlatList-onEndReached'),
+    category: 'ListView',
+  },
+  {
+    key: 'ImageExample',
+    category: 'Basic',
+    module: require('../examples/Image/ImageExample'),
   },
```
Comment (NickGerleman) [inline packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:47]: Why is this removed? It looks like this was a custom windows page.

**Comment by NickGerleman** [comment]
Looking through the upstream commits, I think we might have a new native constant `localeIdentifier` in `NativeI18nManager`. Is this something we should add?

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 1 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:47]: Not removed - I moved it 5 lines above to align with core order and stop getting conflicts for this example.

**Review by NickGerleman — APPROVED** [review]
Overall looks good, but see comments about restoring the Windows custom DatePicker page, along with whether we need to do anything with a new native constant.

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js [hunk 2]
```diff
@@ -26,9 +26,24 @@ const ComponentExamples: Array<RNTesterExample> = [
     module: require('../examples/FlatList/FlatListExample'),
   },
   {
-    key: 'DatePickerExample',
-    category: 'UI',
-    module: require('../examples-win/DatePicker/DatePickerExample'),
```
Comment (asklar) [inline packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: Unintentionally removed?

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 1 above)
Comment (NickGerleman) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:47]: I still don't see DatePickerExample anywhere

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 2 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: Removed on purpose - this page is not in core, and there's no need to demo a deprecated module - we have the community implementation, showcased in the Gallery app.

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 1 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:47]: The comment is on the ImageExample line. For DatePicker,see Alex's comment and my response

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 2 above)
Comment (NickGerleman) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: If we haven't removed the module yet, I'm not sure we should remove the example page and disable the test to visit it. Esp with lean-core in an indeterminate state.

If we did want to remove this though, we should also remove the `DatePickerExample` file this references,

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js [hunk 3]
```diff
@@ -26,9 +26,24 @@ const ComponentExamples: Array<RNTesterExample> = [
...
+  {
+    key: 'FlatList-onViewableItemsChanged',
+    module: require('../examples/FlatList/FlatList-onViewableItemsChanged'),
+    category: 'ListView',
+  },
+  {
+    key: 'FlatList-onEndReached',
+    module: require('../examples/FlatList/FlatList-onEndReached'),
+    category: 'ListView',
+  },
```
Comment (NickGerleman) [inline packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:42]: Can we add this to the `visitAllPages` test which hardcodes the list of RNTester pages to visit?

There's a separate issue for making that dynamic so they don't get out of date.

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 2 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: Since the example is not in core, but the module still is, I'd say let's align with core - so I'd remove the example file altogether, which I forgot. 
I don't feel too strongly about this, though.

**Comment by rectified95** [comment]
@NickGerleman Interestingly, the constant is already in `I18nManagerModule.cpp`
https://github.com/microsoft/react-native-windows/blob/a0e1e995e2363a24102df68577b9435b789968fe/vnext/Microsoft.ReactNative/Modules/I18nManagerModule.cpp#L72

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 2 above)
Comment (NickGerleman) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: Philosophically I think test coverage is good to have for anything we're offering support for. I think those benefits might outweigh loss of alignment.

I refreshed myself a bit on `DatePicker`. I had forgotten that upstream had `DatePickerIOS` and `DatePickerAndroid`, where we expose our own custom `DatePicker` (not exposed by core). We've had it yellow-boxing in at least 0.64, and it doesn't have a pure upstream equivalent, so we could actually probably fully remove it now.

**Comment by NickGerleman** [comment]
> @NickGerleman Interestingly, the constant is already in `I18nManagerModule.cpp`
> https://github.com/microsoft/react-native-windows/blob/a0e1e995e2363a24102df68577b9435b789968fe/vnext/Microsoft.ReactNative/Modules/I18nManagerModule.cpp#L72

Ooh neat. It might have been that upstream was refining flow types to capture it. Always great to discover we don't have extra work to do 😂.

File: packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js (hunk 2 above)
Comment (rectified95) [reply packages/@react-native-windows/tester/src/js/utils/RNTesterList.windows.js:31]: Going to restore the example for now. The whole module and its references can be removed later on.

**Review by asklar — APPROVED** [review]
signing off - are the newly discovered issues regressions on our side that we're only finding now, or regressed by this new ingestion? either way it would be good to tag them accordingly so we can track.
